arwen/context/arwen.go

```diff
@@ -138,13 +138,16 @@ func (host *vmContext) RunSmartContractCreate(input *vmcommon.ContractCreateInpu
 	host.scAddress = address
 	host.addTxValueToSmartContract(input.CallValue, address)
 
-	initialCreateCost := host.computeInitialCreateCost(input)
-
-	if input.GasProvided < initialCreateCost {
+	host.vmInput.GasProvided, err = host.computeInitialCodeCost(
+		input.GasProvided,
+		input.ContractCode,
+		host.GasSchedule().ElrondAPICost.CreateContract,
```

Comment:
In `CreateNewContract()` below, there is a comment saying "creation costs different for ethereum and elrond". But where is this distinction made?

Reply:
ethereumEI and ElrondEI.